static/panes/pane.interfaces.ts

```diff
@@ -24,23 +24,24 @@
 
 import * as monaco from 'monaco-editor';
 
-export interface PaneCompilerState {
-    compilerId: number;
-    compilerName: string;
-    editorId: number;
-}
-
+/**
+ * The base state of a pane as encoded in the URL.
+ *
+ * Be aware this state, and any derived state is part of the public API of
+ * Compiler Explorer, so don't rename or add anything without careful thought.
+ */
 export interface PaneState {
```

Comment:
I think especially when overriding, and its still just data, this should probably be a class and not an interface

Reply:
Same with MonacoPaneState

Reply:
Works for me. It was an `interface` before so I kept that but didn't understand the difference.

Reply:
is it OK to have `interfaces` in the `interfaces.ts` file?